### PR TITLE
Manage IoTProject in the enmasse-operator

### PR DESCRIPTION
**Pull request by lulf**
### Type of change

- Refactoring

### Description

Now that enmasse-operator supports working with both global and local resources, manage IoTProjects in the same process and remove iot-operator deployment to simplify the operational aspect.  Should the need arise in the future, we can split it out again depending on how the IoT approach to shared infra and multiple IoTConfig instances will be.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by lulf** [comment]
@enmasse-ci run tests profile=iot

**Comment by lulf** [comment]
@enmasse-ci run tests profile=iot testcase=MultipleProjectsTest

**Comment by lulf** [comment]
@enmasse-ci run tests profile=iot testcase=MultipleProjectsTest

**Comment by lulf** [comment]
@enmasse-ci run tests profile=iot

File: pkg/controller/iotconfig/project_operator.go [hunk 1]
```diff
@@ -1,84 +0,0 @@
...
-
-func (r *ReconcileIoTConfig) processProjectOperator(ctx context.Context, config *iotv1alpha1.IoTConfig) error {
-	return r.processDeployment(ctx, "iot-operator", config, false, r.reconcileProjectOperator)
-}
-
-func (r *ReconcileIoTConfig) reconcileProjectOperator(config *iotv1alpha1.IoTConfig, deployment *appsv1.Deployment) error {
-
-	install.ApplyDeploymentDefaults(deployment, "iot", deployment.Name)
-
-	applyDefaultDeploymentConfig(deployment, config.Spec.ServicesConfig.Operator.ServiceConfig, nil)
```
Comment (k-wall) [inline pkg/controller/iotconfig/project_operator.go]: For the upgrade case, I see nothing that will remove these resource (iot-operator deployment, service account etc) so they will be orphaned.

**Comment by kornys** [comment]
@enmasse-ci  run tests profile=iot

**Comment by lulf** [comment]
@enmasse-ci run tests profile=iot

**Comment by lulf** [comment]
@enmasse-ci run tests profile=iot

**Comment by lulf** [comment]
@enmasse-ci run tests profile=iot testcase=MqttAdapterTest

**Comment by lulf** [comment]
@enmasse-ci run tests profile=iot

**Comment by lulf** [comment]
@enmasse-ci run tests profile=iot

**Pull request merged**